nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -4,9 +4,21 @@ with lib;
 
 let
   cfg = config.services.matrix-synapse;
+  pg = config.services.postgresql;
+  user = "matrix-synapse";
+  group = user;
+  usePostgresql = cfg.database_type == "psycopg2";
   logConfigFile = pkgs.writeText "log_config.yaml" cfg.logConfig;
   mkResource = r: ''{names: ${builtins.toJSON r.names}, compress: ${boolToString r.compress}}'';
   mkListener = l: ''{port: ${toString l.port}, bind_address: "${l.bind_address}", type: ${l.type}, tls: ${boolToString l.tls}, x_forwarded: ${boolToString l.x_forwarded}, resources: [${concatStringsSep "," (map mkResource l.resources)}]}'';
+  # TODO: finish this and put it in lib somewhere
+  toYaml = value:
```

Comment:
Isn't yaml a superset of JSON? That's how we're getting away with `toJSON` elsewhere.